gherkin/java/src/main/java/gherkin/GherkinDialect.java

```diff
@@ -66,4 +67,26 @@ public GherkinDialect(String language, Map<String, List<String>> keywords) {
     public String getLanguage() {
         return language;
     }
+
+    public String getName() {
+        return (String) (Object) keywords.get("name");
```

Comment:
I haven't seen this kind of double casting before. Can you explain?

Reply:
Not really. I haven't seen it myself before either.

I expected that the result from `keywords.get("name");` would be a list. It turns out that it isn't. It's a String. But as it is declared as a `List<String>`, the compiler will not allow me to cast it to a `String`. My first attempt was divided on two lines and casts. Then I inlined the result and ended up here. I was puzzled for a while and then I gave up. I blaim the internals of GSON.

The attached screenshot is from when I looked at af with a debugger.


![screen shot 2017-12-07 at 10 10 09](https://user-images.githubusercontent.com/541840/33707257-24c75770-db37-11e7-990e-823e988cc911.png)

Reply:
There reason this happens is because of an unchecked cast in GherkinDialectProvider which forces the type to be a  `Map<String, Map<String, List<String>>>`. 

However the data structure is not actually a `Map<String, Map<String, List<String>>>` but  `Map<String, Map<String,  Object>>`  as the properties of a language in `gherking-languages.json` are either a list or a string.

This can be avoided by using the `fromJson` that accepts a type as input and using  `Map<String, Map<String,  Object>>` as type.